src/screenshot.cpp

```diff
@@ -820,7 +820,7 @@ static void HeightmapCallback(void *userdata, void *buffer, uint y, uint pitch,
 	while (n > 0) {
 		TileIndex ti = TileXY(MapMaxX(), y);
 		for (uint x = MapMaxX(); true; x--) {
-			*buf = 256 * TileHeight(ti) / (1 + _settings_game.construction.max_heightlevel);
+			*buf = 256 * TileHeight(ti) / (1 + _settings_game.construction.map_height_limit);
```

Comment:
Should this also be "heightmap_height"?

Reply:
No. `heightmap_height` is only set when you load a heightmap. You can make a heightmap out of any game. So this should scale to the map height limit, I think.

Alternatively, which might be better, it should find the highest point and scale to that.

Reply:
Some more thinking later, and the alternative idea is not really important. When loading it, it will scale to the highest value anyway, so it is just more complex code for no gain, I think :)